core/src/test/scala/unit/kafka/server/LeaderElectionTest.scala

```diff
@@ -86,6 +86,10 @@ class LeaderElectionTest extends ZooKeeperTestHarness {
     val leader2 = waitUntilLeaderIsElectedOrChanged(zkClient, topic, partitionId,
                                                     oldLeaderOpt = if(leader1.get == 0) None else leader1)
     val leaderEpoch2 = ZkUtils.getEpochForPartition(zkClient, topic, partitionId)
+
+    // A LeaderAndIsrRequest with a smaller isr should cause the isrShrinkRate meter to be counted
+    assertEquals("Isr must have shrunk on the replica", 1, servers.head.replicaManager.isrShrinkRate.count())
```

Comment:
Couple of comments on this test:
- The (pre-existing) comment above seems inaccurate - i.e., the last server is not the preferred replica. That said I don't think it matters to this test so we should probably just remove that comment.
- The bigger issue though is that if the server 0 happened to be the follower and then becomes the leader, I think this assert will fail because `inSyncReplicas` is set to empty on follower transition. Can you double-check? If so, we may end up with transient test failures.

Finally, can you rebase?